secp256k1-sys/vendor-libsecp.sh

```diff
@@ -46,28 +96,25 @@ patch "$DIR/src/scratch_impl.h" "./scratch_impl.h.patch"
 patch "$DIR/src/util.h" "./util.h.patch"
 
 # Prefix all methods with rustsecp and a version prefix
-find "$DIR" -not -path '*/\.*' -type f -print0 | xargs -0 sed -i "/^#include/! s/secp256k1_/rustsecp256k1_v${VERSIONCODE}_/g"
-
+find "$DIR" \
+    -not -path '*/\.*' \
+    -not -name "CHANGELOG.md" \
+    -type f \
+    -print0 | xargs -0 sed -i "/^#include/! s/secp256k1_/rustsecp256k1_v${SECP_VENDOR_VERSION_CODE}_/g"
 # special rule for a method that is not prefixed in libsecp
-find "$DIR" -not -path '*/\.*' -type f -print0 | xargs -0 sed -i "/^#include/! s/ecdsa_signature_parse_der_lax/rustsecp256k1_v${VERSIONCODE}_ecdsa_signature_parse_der_lax/g"
-
-# TODO: can be removed once 496c5b43b lands in secp-zkp
-find "$DIR" -not -path '*/\.*' -type f -print0 | xargs -0 sed -i 's/^const int CURVE_B/static const int CURVE_B/g'
-
-while true; do
-    read -r -p "Update Rust extern references and Cargo.toml as well? [yn]: " yn
-    case $yn in
-        [Yy]* ) break;;
-        [Nn]* ) exit;;
-        * ) echo "Please answer yes or no.";;
-    esac
-done
-
-cd "$ORIGDIR"
+find "$DIR" \
+    -not -path '*/\.*' \
+    -type f \
+    -print0 | xargs -0 sed -i "/^#include/! s/ecdsa_signature_parse_der_lax/rustsecp256k1_v${SECP_VENDOR_VERSION_CODE}_ecdsa_signature_parse_der_lax/g"
 
+cd "$SECP_SYS"
 # Update the `links = ` in the manifest file.
-sed -i -r "s/^links = \".*\"$/links = \"rustsecp256k1_v${VERSIONCODE}\"/" Cargo.toml
-
+sed -i -r "s/^links = \".*\"$/links = \"rustsecp256k1_v${SECP_VENDOR_VERSION_CODE}\"/" Cargo.toml
 # Update the extern references in the Rust FFI source files.
-find "./src/" -name "*.rs" -type f -print0 | xargs -0 sed -i -r "s/rustsecp256k1_v[0-9]+_[0-9]+_[0-9]+_(.*)([\"\(])/rustsecp256k1_v${VERSIONCODE}_\1\2/g"
+find "./src/" \
+    -name "*.rs" \
+    -type f \
+    -print0 | xargs -0 sed -i -r "s/rustsecp256k1_v[0-9]+_[0-9]+_[0-9]+_(.*)([\"\(])/rustsecp256k1_v${SECP_VENDOR_VERSION_CODE}_\1\2/g"
+
+popd > /dev/null
 
```

Comment:
Bash question: Any reason not to `exit 0` at the end of the script?

Reply:
We implicitly return the exit value of the last executed command. With `set -e` this means "nonzero if anything failed, 0 otherwise".

I can add an `exit 0` but it'd be purely redundant.

Reply:
Fair points, thanks. I didn't think of the implications of `set -e` for the return value.